lib/cloud-watch-event-listener.js

```diff
@@ -194,81 +212,15 @@ class CloudWatchEventListener extends events.EventEmitter {
         let delay = dbResponse.lastEvent - generated;
         this.monitor.measure('global.cwe-out-of-order.delay', delay);
         this.monitor.measure(`${region}.cwe-out-of-order.delay`, delay);
-        log.info({region, id, state}, 'CloudWatch Event delivered out of order');
-      }
-    } else if (dbResponse.length === 0) {
-      // We don't want to use the same transaction because there's no need.  We
-      // don't want to hold the lock longer than needed
-      await this.state.commitTransaction(transaction);
-
-      if (state === 'pending' || state === 'running') {
-        // For events which are pending or running but which aren't in the
-        // instances table already we'll use the describeInstances table to look
-        // up the information about them.  This is less efficient, but still
-        // works great.
-        let apiResponse;
-        try {
-          apiResponse = await this.runaws(this.ec2, 'describeInstances', {
-            InstanceIds: [id],
-          });
-        } catch (err) {
-          // We're ignoring this error because it might happen that it is only
-          // delay in internal EC2 updates.  Given that, we're going to wait
-          // until we've exhausted all redeliveries, which we do in the dead
-          // letter queue handler
-          if (err.code !== 'InvalidInstanceID.NotFound') {
-            this.monitor.reportError(err);
-          } else {
-            this.monitor.count('global.api-lag', 1);
-            this.monitor.count(`${region}.api-lag`, 1);
-          }
-          throw err;
-        }
-
-        // TODO: CRITICAL Skip things which have a keyname which does not match ours
-
-        assert(Array.isArray(apiResponse.Reservations));
-        assert(apiResponse.Reservations.length === 1);
-        assert(Array.isArray(apiResponse.Reservations[0].Instances));
-        assert(apiResponse.Reservations[0].Instances.length === 1);
-        let instance = apiResponse.Reservations[0].Instances[0];
-
-        let [provisionerId, workerType] = instance.KeyName.split(':');
-        let instanceType = instance.InstanceType;
-        let az = instance.Placement.AvailabilityZone;
-        let imageId = instance.ImageId;
-        let launched = new Date(instance.LaunchTime);
-
-        // We check for workertype being truthy because it's always possible that
-        // this instance is one which is not in the provisioner/ec2-manager sphere
-        // of knowledge at all and as such has no colons in its name.
-        if (workerType && missingTags(instance)) {
-          await this.tagger.tagResources({
-            ids: [id],
-            workerType: workerType,
-            region: region,
-          });
-        }
-        let opts = {
-          workerType,
+        log.info({
```

Comment:
If this is something we want to watch, should it be a sentry warning or even just a sentry info?